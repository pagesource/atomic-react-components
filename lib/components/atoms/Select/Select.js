// @flow
import React from 'react';
import type { Node, ComponentType } from 'react';
import styled from 'styled-components';
import styles from './Select.style';

type Props = {
  id: string,
  className?: string,
  name: string,
  disabled?: string | boolean,
  elementLocator?: string,
  options: any,
  selectedOption?: string,
  placeholder?: Node,
  ref?: ?HTMLSelectElement,
};

const Select = React.forwardRef<Props, HTMLSelectElement>(
  (
    {
      id,
      className,
      disabled,
      elementLocator,
      placeholder,
      options,
      selectedOption,
      name,
      ...others
    }: Props,
    ref
  ): Node => (
    <select
      {...others}
      id={id}
      name={name}
      className={className}
      disabled={disabled}
      data-locator={elementLocator || `select-input-${name}-${id}`}
      ref={ref}
    >
      {placeholder && <option value="">{placeholder}</option>}
      {options.map((opt) => (
        <option key={opt} value={opt}>
          {opt}
        </option>
      ))}
    </select>
  )
);

// $flow-disable-line  eslint-disable-line
Select.defaultProps = {
  disabled: false,
  selectedOption: '',
  placeholder: '',
  className: '',
  elementLocator: '',
  ref: null,
};

const StyledSelect: ComponentType<Props> = styled(Select)`
  ${styles};
`;

export default StyledSelect;
export { Select as SelectVanilla };
