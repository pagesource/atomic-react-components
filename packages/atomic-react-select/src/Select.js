import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import styles from './Select.style';

const Select = React.forwardRef(
  (
    {
      id,
      className,
      disabled,
      elementLocator,
      placeholder,
      options,
      selectedOption,
      name,
      ...others
    },
    ref
  ) => (
    <select
      id={id}
      name={name}
      className={className}
      disabled={disabled}
      data-locator={elementLocator || `select-input-${name}-${id}`}
      ref={ref}
      {...others}
    >
      {placeholder && <option value="">{placeholder}</option>}
      {options.map((opt) => (
        <option key={opt} value={opt}>
          {opt}
        </option>
      ))}
    </select>
  )
);

Select.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  name: PropTypes.string,
  disabled: PropTypes.string,
  elementLocator: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectedOption: PropTypes.string,
  placeholder: PropTypes.node,
  ref: PropTypes.element,
};

Select.defaultProps = {
  disabled: false,
  selectedOption: '',
  placeholder: '',
  className: '',
  elementLocator: '',
  name: '',
  ref: null,
};

const StyledSelect = styled(Select)`
  ${styles};
`;

export default StyledSelect;
export { Select as SelectVanilla };
