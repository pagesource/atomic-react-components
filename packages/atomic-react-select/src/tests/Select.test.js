import React from 'react';
import { shallow, render } from 'enzyme';
import { SelectVanilla } from '../Select';

describe('Select Component', () => {
  let input = '';
  const props = {
    id: 'dropdown-1',
    name: 'my-drop',
    elementLocator: 'my-custom-select-locator',
    onChange: () => {},
  };

  test('should render correctly', () => {
    input = shallow(<SelectVanilla options={['Option 1', 'Option 2']} {...props} />);
    expect(input).toMatchSnapshot();
  });
  test('should check the data attribute of dropdown/select', () => {
    const select = render(<SelectVanilla options={['Option 1', 'Option 2']} {...props} />);

    expect(select.attr('data-locator')).toEqual(`${props.elementLocator}`);
  });
});
