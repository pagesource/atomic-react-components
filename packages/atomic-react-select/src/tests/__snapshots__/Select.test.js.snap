// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component should render correctly 1`] = `
<select
  className=""
  data-locator="my-custom-select-locator"
  disabled={false}
  id="dropdown-1"
  name="my-drop"
  onChange={[Function]}
>
  <option
    key="Option 1"
    value="Option 1"
  >
    Option 1
  </option>
  <option
    key="Option 2"
    value="Option 2"
  >
    Option 2
  </option>
</select>
`;
